packages/vant/src/tabs/Tabs.tsx

```diff
@@ -445,6 +446,18 @@ export default defineComponent({
       }
     );
 
+    watch(
+      () => root.value,
+      (el) => {
+        const isSupported = window && 'ResizeObserver' in window;
+        if (isSupported && el) {
+          observer = new ResizeObserver(setLine);
```

Comment:
How about using the [useVisibilityChange](https://github.com/youzan/vant/blob/main/packages/vant/src/composables/use-visibility-change.ts) hook?

Reply:
yeah, I didn't know about this hook at first, I'll modify it immediately.